build_tools/check_backticks.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+"""Test script to check for invalid use of single-backticks."""
+
+__author__ = ["geetu040"]
+
+import ast
+import os
+import re
+
+
+def find_py_files(folder_path):
+    """Find all Python files in a given folder path."""
+    py_files = []
+    for root, _, files in os.walk(folder_path):
+        for file in files:
+            if file.endswith(".py"):
+                py_files.append(os.path.join(root, file))
+    return py_files
+
+
+def extract_docstrings(filename):
+    """Extract docstrings from a Python file."""
+    with open(filename) as file:
+        tree = ast.parse(file.read(), filename=filename)
+
+    docstrings = {}
+    for node in ast.walk(tree):
+        if isinstance(node, (ast.FunctionDef, ast.ClassDef, ast.Module)):
```

Comment:
A node of type `ast.Module` represents the file itself and is a root node of the tree.

`Enums` are basically python classes that inherit `enum.Enum` class, so that is covered as well

*the updated code deals this differently*